relay.js

```diff
@@ -178,6 +199,9 @@ export function relayConnect(url, onNotice = () => {}, onError = () => {}) {
   return {
     url,
     sub,
+    async publishWithConfirmation(event) {
+      return trySendWithConfirmation(['EVENT', event])
+    },
```

Comment:
I think it is better to nuke the `publish` method and replace it with this new approach that returns a Promise.

but there is a catch: the Promise cannot rely solely on the `OK` message. it must wait for the `OK` message, but if that doesn't happen it must do as the current `publish()` does: subscribe to the event and wait for it to be returned -- then if all of that fails after a while call the `reject` callback of the promise.

what do you think?

Reply:
I really dont understand the first approach so i did the new one for dont messing it up. I think it is a good solution.
Can u explain me the first publish function? what are the callbacks and the status? What is expecting the sub created? only the same event? It is a kind of confirmation out of the nips? Maybe is write an event and then trying to read it for confirm?